Refactor html output

jonathanyeong opened the pull request:
This pull request takes out the static html from trello_newsletter.rb and puts it into html_stitcher.rb. It also adds some tests to make sure index.html is structurally sound (has head and body tags).

jonathanyeong committed:
Add nokogiri gem to project

This gem will be used to test certain html elements exist in the
index.html file.

jonathanyeong committed:
Implement initial tests for index.html

This test will check if the head and body tags are in the html.

jonathanyeong committed:
Call html stitcher to generate head for template

The head html stitcher will replace the large head string.

jonathanyeong committed:
Add html_stitcher class to project

The html stitcher will contain all the static html that is used to
generate the index.html page.

jonathanyeong committed:
Remove template html and call html_stitcher methods

jonathanyeong committed:
Add static html to the html stitcher class

jonathanyeong committed:
Remove comment

at_css method doesn't apply anymore

sarony commented on lib/html_stitcher.rb:
```diff
@@ -0,0 +1,679 @@
+class HtmlStitcher
+  def head
+    head_string = <<-DOC.gsub(/^ {4}/, '')
```
Why do we need to set this local variable?

sarony commented on lib/html_stitcher.rb:
```diff
@@ -0,0 +1,679 @@
+  end
+
+  def body_start(meta)
+    body_string = <<-DOC.gsub(/^ {4}/, '')
```
Same comment as above. Why the local variable?

sarony commented on lib/html_stitcher.rb:
```diff
@@ -0,0 +1,679 @@
+  end
+
+  def body_end
+    body_string = <<-DOC.gsub(/^ {4}/, '')
```
Same comment as above on the local variable

sarony commented on spec/views/index.html_spec.rb:
```diff
@@ -0,0 +1,17 @@
+require 'nokogiri'
+
+describe "html output" do 
```
good use of Nokogiri!

sarony commented:
Let me know about those local variables, I don't think we need them. Otherwise, looks good!

jonathanyeong commented:
I'm using the local variables to store the heredoc string and return it when the html_stitcher is called. I wasn't sure if it would still return the heredoc string if you take the local variables out. I'll see if it works.

jonathanyeong committed:
WIP: Adds html content method

Moves the code from trello_newsletter.rb to html_factory.rb

TODO
- Finish content method. Uncomment out line 718 and move it into a
  Heredoc string.

jonathanyeong committed:
Add the rest of the newsletter to the html factory

This code was previously commented out. Uncomment this code and add it
to the Heredoc statement. This commit will also fix formatting issues
such as unnecessary double quotes and back slashes.

jonathanyeong committed:
Add a vcr cassette for testing newsletter generation

jonathanyeong commented:
@sarony updated the local variables. Finished adding the rest of the newsletter to the HTML factory. Let me know if I can change anything when we meet next.
